Feature/200 settings dialog ux

davemfish opened the pull request:
Fixes #200 

In addition to the issues mentioned in the issue, this PR also switches the `n_workers` input field to a dropdown menu populated with numbers from -1 to the number of available CPUs. That data comes from NodeJS and is sent to the renderer at app startup. That removes the need to validate that input field.

I also added some expandable/collapsible helptext for that parameter, similar to what we have in the Qt UI. 

![settings_ui](https://user-images.githubusercontent.com/4071255/142479397-2b89bcb8-43a9-43e6-b135-f3d6b586d0a0.PNG)

davemfish commented:
Sorry for the future merge conflicts @emlys since you already changed some of this stuff in your branch!

emlys commented on src/renderer/components/SettingsModal/index.jsx:
```diff
@@ -1,98 +1,68 @@
-        nWorkers: '',
-        loggingLevel: '',
-        sampleDataDir: null,
-      },
```
I like that we're no longer having a separate copy of the settings in this component state!

emlys reviewed (changes requested):
@davemfish, I like the dropdown for the n workers value! To further streamline it, instead of having the options '-1' and '0' in the dropdown, then defining them in the collapsible box, what if the definitions were right in the dropdown? So the options would be like "Synchronous (recommended)", "Multithreaded single process", "2 processes", "3 processes", ...

emlys commented:
> Sorry for the future merge conflicts @emlys since you already changed some of this stuff in your branch!

No worries!

I'm not sure why that test is failing... looks like it's not successfully getting the args specs?

davemfish commented:
> > Sorry for the future merge conflicts @emlys since you already changed some of this stuff in your branch!
> 
> No worries!
> 
> I'm not sure why that test is failing... looks like it's not successfully getting the args specs?

Yeah that was weird. It looks like it failed to download the invest binaries due to a weird ssl error. But that reminded me to update the invest version as well, so let's wait for everything to re-run.

davemfish commented:
> @davemfish, I like the dropdown for the n workers value! To further streamline it, instead of having the options '-1' and '0' in the dropdown, then defining them in the collapsible box, what if the definitions were right in the dropdown? So the options would be like "Synchronous (recommended)", "Multithreaded single process", "2 processes", "3 processes", ...

Yes, good suggestion. I switched the dropdown to use the same text that we have in the Qt UI. I modified the extra text, I still think it's useful to have a little extra info available than fits in the label. And I tweaked the layout to accommodate the wider dropdown content. Back to you @emlys 

![settings_ui](https://user-images.githubusercontent.com/4071255/142501562-ae59e53b-29f1-4529-acf2-40e94c0c95c7.PNG)

emlys reviewed (approved):
Looks good!
